### PR TITLE
Nicer syntax for ci.yml matrix env variables

### PR DESCRIPTION
**Pull request by ewels**
Continuing from https://github.com/nf-core/tools/pull/1343, refactoring to try to make this logic a little easier to read / understand

## PR checklist

 - [x] This comment contains a description of changes (with reason)
 - [ ] `CHANGELOG.md` is updated

### CODE REVIEW

File: nf_core/pipeline-template/.github/workflows/ci.yml [hunk 1]
```diff
@@ -18,21 +18,28 @@ jobs:
...
       - name: Check out pipeline code
         uses: actions/checkout@v2
 
       - name: Install Nextflow
         env:
           CAPSULE_LOG: none
+          {% raw %-}
+          NXF_VER: ${{ matrix.NXF_VER }}
+          NXF_EDGE: ${{ matrix.NXF_EDGE }}
+          {-% endraw %}
```
Comment (ewels) [inline nf_core/pipeline-template/.github/workflows/ci.yml:39]:
```suggestion
          {% raw -%}
          NXF_VER: ${{ matrix.NXF_VER }}
          NXF_EDGE: ${{ matrix.NXF_EDGE }}
          {%- endraw %}
```

**Comment by ewels** [comment]
Sigh, linting code needs updating. Not sure if it's really worth it..

@ggabernet you said the existing syntax was broken currently, would you mind checking if this refactored version works? If so then I might be bothered to update the linting here..

File: nf_core/pipeline-template/.github/workflows/ci.yml [hunk 2]
```diff
@@ -18,21 +18,28 @@ jobs:
...
     steps:
       - name: Check out pipeline code
         uses: actions/checkout@v2
 
       - name: Install Nextflow
         env:
           CAPSULE_LOG: none
+          {% raw -%}
+          NXF_VER: ${{ matrix.NXF_VER }}
+          NXF_EDGE: ${{ matrix.NXF_EDGE }}
```
Comment (ewels) [inline nf_core/pipeline-template/.github/workflows/ci.yml:38]: x-ref https://github.com/nextflow-io/nextflow/issues/2467
```suggestion
          # NXF_EDGE: ${{ matrix.NXF_EDGE }}
```

**Comment by ggabernet** [comment]
Ah no it was not broken, I still had an older nf-core/tools `dev` version installed. Like it is now on the `dev` branch works for me now, sorry for the confusion.

And this proposal you are making does not work for me, unfortunately: https://github.com/ggabernet/elixir-workshop/actions/runs/1508733244

**Review by ggabernet — APPROVED** [review]
Looks good to me!

This solution also works for me here: https://github.com/ggabernet/elixir-workshop/runs/4337246764?check_suite_focus=true

**Review by grst — APPROVED** [review]
I approve!

**Comment by drpatelh** [comment]
Lots of solutions but the linting is the real boss 🤣

**Review by drpatelh — APPROVED** [review]
I approve 3

**Pull request merged**